Doc/library/ssl.rst

```diff
@@ -1210,6 +1212,20 @@ SSL sockets also have the following additional methods and attributes:
    .. versionchanged:: 3.9
       IPv6 address strings no longer have a trailing new line.
 
+.. method:: SSLSocket.get_verified_chain()
+
+   Returns verified certificate chain provided by the other
+   end of the SSL channel as a list of DER-encoded bytes.
+
+   .. versionadded:: 3.13
+
+.. method:: SSLSocket.get_unverified_chain()
+
+   Returns unverified certificate chain provided by the other
```

Comment:
For documentation purposes, it'd be good to describe when someone would want to call one of these vs the other or when it makes sense to call get_verified_chain vs get_unverified_chain.  Its is somewhat unusual to have APIs like this that seem to require prior knowledge of the state of the connection before knowing which one to use (I do think the caller _should_ know that state given the importance when using TLS).

At the very least, answer these potential questions in the docs for each method:

When a TLS connection is made with certificate verification enabled, what happens when get_unverified_chain is called?

When a TLS connection is made _without_ certificate verification enabled, what happens when get_verified_chain is called?

Reply:
Ok, let me put these answers.

Reply:
Actually, it is not that easy for me to prepare this doc, because honestly, I don't know who and for what someone would like to use `get_verified_chain`.

Right now using `get_unverified_chain` gives me the possibility to perform OCSP checks for every certificate-issuer pair that was presented in TLS handshake. The `get_verified_chain` returns a shorter list, because some of the certificates are dropped during verification. Maybe at some point, I will switch to `get_verified_chain` because there will be no point in verifying the raw chain. I don't know, I can only speak about myself.